### ISSUE TITLE
Duplicate 

### ISSUE THREAD

**Issue by JesseLeite** [jenssegers/laravel-rollbar#61]
Your [usage documentation](https://github.com/jenssegers/laravel-rollbar#usage) suggests using `Log` facade to log all exceptions for Rollbar.  ie)
```php
public function report(Exception $e)
{
    \Log::error($e);
    return parent::report($e);
}
```
However, the `parent::report($e)` call there also logs exceptions.  This means that reportable exceptions are being logged twice.  Can we not just let the `parent::report($e)` method do the logging?

**Comment by gregrobson** [comment]
This appears to be an issue with Laravel, having looked at.
https://github.com/laravel/framework/blob/5.2/src/Illuminate/Foundation/Exceptions/Handler.php

`parent::report(Exception $e)` is calling this parent method.

```php
    public function report(Exception $e)
    {
        if ($this->shouldReport($e)) {
            $this->log->error($e);
        }
    }
```

`$this->log` should be an implementation of `LoggerInterface` and the signature should be `error(Exception $e, array $context = [])`

There's no way to pass contact through Laravel's implementation. I'll post up an issue later today. For the moment, it might be best to check whether Rollbar is enabled in your current environment and then decide whether to call `\Log::error($e)` or `parent::report($e)`

**Comment by tommymarshall** [comment]
This is still occurring even though [Laravel has updated this method](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Foundation/Exceptions/Handler.php#L59). Any other ideas what's going on? I am on 5.3.28.

**Comment by sweet-greg** [comment]
@gregrobson It looks like Laravel does not see this an error from their side. How did you resolve this? Is it safe to drop `\Log::error($e);`? I'm seeing 95% of exceptions as duplicates, but sometimes they still come as one... Which seems odd to me...

**Comment by JesseLeite** [comment]
@sweet-greg the exceptions that only report once in rollbar are probably in your `$dontReport` array in `app/Hander.php`.  The exceptions that report twice are _not_ in your `$dontReport` array, so they are being reported by the `parent::report()` as well as your `\Log::error($e)` call.  That's my assumption anyway.

If that's the case, I think it would be much more intuitive to just let `parent::report()` do it's thing, and not have rollbar users add the extra `\Log::error($e)`.  If I don't want to report a specific kind of exception, like a form ValidationException, then I likely don't want that in Rollbar either.  Every time someone fills out a form and forgets a field, that's not an issue I need to deal with as a developer.

Thoughts?

**Comment by sweet-greg** [comment]
@JesseLeite Thanks for getting back at me, if you check out `parent::report()` the same check is done (although in reverse). So that should not be the reason for sometimes having no duplicates. 

    // Illuminate\Foundation\Exceptions\Handler@report
    if ($this->shouldntReport($e)) {
        return;
    }

**Comment by JesseLeite** [comment]
That's exactly what I was getting at, but in much less words 😃   `parent::report()` respects your `$dontReport` array, and I think Rollbar should follow suit and respect your `$dontReport` array.  The extra `\Log::error($e)` is confusing things for users of this package, and causing the duplicates.

**Comment by rdrnnr87** [comment]
@gregrobson I agree with the issue you submitted to Laravel repo.  It seems like @GrahamCampbell may have misunderstood what you were requesting.  We want to pass the error context to `parent::report($e)` and then we can remove `\Log::error($e, $context)` entirely.

**Comment by samgavinio** [comment]
This is configured in the service provider to listen to `illuminate.log` events. This is only fired when the Log facade (wrapper for [\Illuminate\Log\Writer](https://github.com/laravel/framework/blob/02444b7450350db17a7607c8a52f7268ebdb0dad/src/Illuminate/Log/Writer.php#L201)) is used. If you don't use the facade you don't get rollbar logging at all with this package.

**Comment by patrick-proulx** [comment]
This is old, but I've found a solution if you're using the Rollbar Service Provider. In `report(Exception $e)`, you can add:
```
if ($user = Auth::user()) {
            app('Rollbar\RollbarLogger')->configure([
                'person' => [
                    'id'       => $user->id,
                    'username' => $user->first_name . ' ' . $user->last_name,
                    'email'    => $user->email,
                ],
            ]);
        }
```
